### PR TITLE
mangle slashes in repo names

### PR DESCRIPTION
**Pull request by krzyzacy**
https://github.com/krzyzacy/test-infra/blob/master/jenkins/bootstrap.py#L684

/shrug
(I think it had never hit into that else before for github...... but......)

/assign @BenTheElder

### CODE REVIEW

**Comment by BenTheElder** [comment]
yeah github repos cannot do this, at least one of our gerrits does allow this though. xref: https://github.com/kubernetes/test-infra/issues/8831

File: prow/gerrit/gerrit.go [hunk 1]
```diff
@@ -310,11 +310,12 @@ func (c *Controller) ProcessChange(change gerrit.ChangeInfo) error {
...
 
 				// mangle
-				// https://gubernator.k8s.io/build/gob-prow/pr-logs/pull/some-repo/8940/pull-test-infra-presubmit//
+				// https://gubernator.k8s.io/build/gob-prow/pr-logs/pull/some/repo/8940/pull-test-infra-presubmit//
 				// to
-				// https://gubernator.k8s.io/builds/gob-prow/pr-logs/pull/some-repo/8940/pull-test-infra-presubmit/
+				// https://gubernator.k8s.io/builds/gob-prow/pr-logs/pull/some_repo/8940/pull-test-infra-presubmit/
 				url = b.String()
 				url = strings.Replace(url, "build", "builds", 1)
+				url = strings.Replace(url, change.Project, strings.Replace(change.Project, "/", "_"), 1)
```
Comment (BenTheElder) [inline prow/gerrit/gerrit.go]: add a `// TODO` about removing this if we ever get a better solution for the paths, maybe xref https://github.com/kubernetes/test-infra/issues/8831

**Comment by stevekuznetsov** [comment]
Can we use disallowed characters as the mangle target? Won't `repo/name` and `repo_name` don't collide?

**Comment by BenTheElder** [comment]
this is just matching how we upload them now so the links work. we should probably consider how to do this better going forward though, probably not using templates :-) related: https://github.com/kubernetes/test-infra/issues/8831

**Comment by krzyzacy** [comment]
@stevekuznetsov or I can mangle repo: repo/name to become repo: repo_name, if you think that's more correct

**Pull request merged**